app/models/sample.rb

```diff
@@ -730,9 +730,15 @@ def private_until
   # Get Metadata objects augmented with MetadataField.base_type
   def metadata_with_base_type
     metadata.map do |m|
-      m.attributes.merge(
-        "base_type" => Metadatum.convert_type_to_string(m.metadata_field.base_type)
-      )
+      resp = m.attributes
+      base_type = m.metadata_field.base_type
+
+      # Special-case for locations (avoid extra call to #metadata_field.base_type in #validated_value)
+      if base_type == Metadatum::LOCATION_TYPE
```

Comment:
Could this mean a bunch of extra calls to location in a bulk operation?

Reply:
it looks like you'd get 1 more call to the Locations table per sample (assuming every sample has 1 Location). This could be batched at the place before metadata_with_base_type is called. Looks like only one call in report_helper#samples_taxons_details is in a batch context, and that one is still making a separate metadata call per sample